config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-3-upgrades.yaml

```diff
@@ -47,17 +46,20 @@ periodics:
       resources:
         requests:
           cpu: 7300m
+          memory: 9Gi
+        limits:
+          cpu: 7300m
+          memory: 9Gi
```

Comment:
@killianmuldoon Do you remember the Slack discussion? I think they were saying we don't have to be too strict with memory. Just want to avoid having to debug flakes becuase of OOM kills
(the Prow nodes are relatively memory heavy)

Reply:
This is the discussion in question - https://kubernetes.slack.com/archives/C7J9RP96G/p1685551306721239

I think we can go to 32 Gi here without causing issus. The instances apparently have 8Gi / core so 4Gi per core is still below utilization.

Reply:
Thx for looking it up. Sound reasonable for me to then use 32 Gi for all e2e / conformance / upgrade job

Reply:
I've updated all e2e / conformance / upgrade jobs to 32Gi